src/memex/models/document.py

```diff
@@ -393,10 +395,14 @@ def merge_documents(session, documents, updated=None):
             m.document = master
             m.updated = updated
 
-        session.delete(doc)
-
     try:
         session.flush()
+        session.query(Annotation) \
+            .filter(Annotation.document_id.in_(duplicate_ids)) \
```

Comment:
Again, this can I think be simplified (and `duplicate_ids` removed) when you update the `document` relationship on `Annotation` to just `Annotation.document.in_(duplicates)`. Maybe worth making a note that says as much.